docs/Contributing/architecture/system_design/cache.md

```diff
@@ -0,0 +1,68 @@
+[Back to top](./README.md)
+# Cache
+
+Fleet uses Redis (standalone or cluster) as an ephemeral coordination and acceleration layer. No authoritative data stored here—safe to flush with minor transient impact.
+
+## Use cases
+* Live query campaigns (active query set + host completion tracking)
+* Distributed locks for cron / async tasks (`redis_lock` namespace)
+* Async host detail queue (when async processing enabled) – hosts push detail payload keys, workers pop
+* Software installer token metadata (short-lived) – speed validation before DB write
+* MDM profile preassignment (Apple) – temporarily associate serials / external ids to team profiles during enrollment handshake
+* Rate limiting / throttling counters (selected endpoints)
+* Caching AppConfig/Teams (optional small TTL to shield bursts)
+
+## Key patterns
+| Purpose | Pattern | Expiration |
+|---------|---------|------------|
+| Live query SQL | `live_query:campaign:<id>:sql` | campaign end / manual cleanup |
+| Active queries set | `live_query:active` (SET) | continuous (cleanup cron) |
+| Host completion | `live_query:campaign:<id>:hosts_completed` (SET) | cleanup on finish |
+| Async host queue | `async_host:detail` (LIST / STREAM) | length limited by workers |
+| Lock | `lock:<name>` (string w/ token) | TTL = lock timeout |
+| MDM preassign | `mdm:apple:preassign:<serial>` | minutes (enrollment window) |
+| Installer token | `sw:installer:token:<uuid>` | short (<=1h) |
```

Comment:
That'd be nice to have documented if it was reliable, but I'm afraid this will quickly go outdated as it's a bit too low-level.